model/src/lib.rs

```diff
@@ -105,12 +105,28 @@ impl OrderCreation {
         address
     }
 }
+pub struct InvalidSignatureError;
 
 // See https://github.com/gnosis/gp-v2-contracts/blob/main/src/contracts/libraries/GPv2Encoding.sol
 impl OrderCreation {
     const ORDER_TYPE_HASH: [u8; 32] =
         hex!("b71968fcf5e55b9c3370f2809d4078a4695be79dfa43e5aa1f2baa0a9b84f186");
 
+    pub fn order_creation_to_full_order(
+        self: OrderCreation,
+    ) -> Result<Order, InvalidSignatureError> {
```

Comment:
I'd like to keep this in the orderbook crate because that's the only place where we do this operation.